### ISSUE TITLE
Assistance w the /help command

### ISSUE THREAD

**Issue by bdmrsmokesganj** [EssentialsX/Essentials#3773]
Plugin List : 
![plugins](https://user-images.githubusercontent.com/73685474/99037878-9a3a7380-2552-11eb-84e4-875bbd187e38.png)
-
-
-
I'm sorry for not following the template but I'm new to this and need a little help
so I have permissions setup w luck perms and my commands show in /help like so:
![Untitled](https://user-images.githubusercontent.com/73685474/99037349-880c0580-2551-11eb-9412-01c134b0834f.png)
but **I am looking to remove the part at the top where it shows "Essentials:  Plugin help: /help essentials" and "EssentialsSpawn: Plugin help: /help essentialsspawn"** 

I also cannot figure out how to hide commands from showing that I never gave perms for. When default players type "/help (space)" they see all kinds of commands that they dont have access to like so:
![Not allowed commands](https://user-images.githubusercontent.com/73685474/99037757-5b0c2280-2552-11eb-9b22-2f0da3f8a3df.png)
and I need to know how to hide those commands from default players so they only see the commands they have perms for.

**Comment by ram1660** [comment]
For the first question, you can't remove this text unless you modify Essentials itself.
For the second question, I think you should look for a plugin that can disable the autocomplete, Essentials can't do that.

**Comment by pop4959** [comment]
You can actually write a `help.txt` file in the /plugins/Essentials/ directory to make it show whatever you want in the /help command. Although let me warn you that this is indeed tedious and you will have to specify all commands you want in it individually. However, of course, that gives you quite a lot of freedom about how you want the help formatted.

Completions for commands players don't have access to should typically not show up, but as ram is saying I don't think this is an issue with Essentials. If need be, you can install a plugin to block specific completions that you wish to avoid, but I'd suggest restarting your server first to see if the completions get reloaded with the correct permissions.

Hope that helps!

**Comment by mdcfe** [comment]
Just to add, you should make sure you haven't given players `essentials.help.*` by mistake - that permission gives players the ability to see all commands, not just ones they have access to.

**Comment by serious-angel** [comment]
> Just to add, you should make sure you haven't given players `essentials.help.*` by mistake - that permission gives players the ability to see all commands, not just ones they have access to.

It's probably also worth to mention that plugins can also relate to the issue. For example, [Plugin Hide Pro](https://www.spigotmc.org/resources/%E2%9C%94%EF%B8%8F-plugin-hide-pro-customize-players-tabcompletion-spigot-bungee-velocity-%E2%9C%94%EF%B8%8F.77112).

> ### How to allow the base command and some subargs? [Only for the commands lists]
> If you are using whitelist mode and want the base command and only some subargs of the command to be executable, then you need the "`~`" after the base command: For example, you have the warp command with the subargs admin, test, player, city Then if you would like to allow "`/warp`", "`/warp player`", and "`/warp city`" to be executable, you would list it like this:
> 
> ```yaml
> commands:
>   - warp ~
>   - warp player
>   - warp city
> ```

[Source](https://github.com/Nononitas/Plugin-Hide-Pro/wiki/FAQ's#how-to-allow-the-base-command-and-some-subargs-only-for-the-commands-lists)

In other words, even if the commands are not permitted nor available, the following might expose command argument previews anyways:

```yaml
groups:
  default:
    commands:
      - help
    tabcomplete:
      - help
```

In contrast, to hide everything:

```yaml
groups:
  default:
    commands:
      - help
    tabcomplete:
      - help ~
```

**Comment by mdcfe** [comment]
@F8ER Please don't bump years-old solved issues. EssentialsX provides this functionality by default as explained above - there's no need to use paid plugins to do the same job.
